lib/modules/address.js

```diff
@@ -202,4 +214,55 @@ AddressModule.prototype.isSpent = function(output, queryMempool, callback) {
   });
 };
 
+AddressModule.prototype.getSpendTxForOutput = function(txid, outputIndex, queryMempool, callback) {
+  var self = this;
+
+  var key = [AddressModule.PREFIXES.SPENTS, txid, outputIndex].join('-');
+  this.db.store.get(key, function(err, spendTxId) {
+    if(err) {
+      return callback(err);
+    }
+
+    self.db.getTransaction(spendTxId, queryMempool, callback);
+  });
+};
+
+AddressModule.prototype.getTransactionsForAddress = function(address, queryMempool, callback) {
+  var self = this;
+
+  this.getOutputs(address, queryMempool, function(err, outputs) {
+    if(err) {
+      return callback(err);
+    }
+
+    var transactions = [];
+
+    async.eachSeries(
+      outputs,
+      function(output, next) {
+        self.db.getTransaction(output.txid, queryMempool, function(err, tx) {
+          if(err) {
+            return next(err);
+          }
+
+          transactions.push(tx);
```

Comment:
There can be several outputs within the same transaction, leading to a transaction being added twice.